web-ng/src/app/components/project-header/project-header.component.ts

```diff
@@ -43,4 +66,21 @@ export class ProjectHeaderComponent implements OnInit {
       autoFocus: false,
     });
   }
+
+  /**
+   * Updates the project title with input element value.
+   *
+   * @param evt the event emitted from the input element on blur.
+   */
+  updateProjectTitle(value: string) {
+    if (!this.projectId) {
+      return Promise.reject(new Error('Project not loaded'));
+    }
+    if (value === this.title) return;
```

Comment:
This would return `undefined`, while other paths return `Promise`. Not sure what the event handler does with the return value, but can we return `Promise.resolve()` here instead for consistency?